### PR TITLE
DS 671 - Input refactor

### PR DESCRIPTION
**Pull request by zuzannanap**

### CODE REVIEW

File: packages/design-system/src/components/Input/Input.interface.ts [hunk 1]
```diff
@@ -1,85 +1,68 @@
...
-  placeholder?: string;
-  isError?: boolean;
-  disabled?: boolean;
-  required?: boolean;
-  suffixVariant?: suffixVariants;
-  prefixVariant?: prefixVariants;
-  ariaLabel?: string;
-  inputRef?: RefObject<HTMLInputElement>;
-  iconAriaLabel?: string;
+  style?: CSSProperties;
```
Comment (piotrekzawada) [inline packages/design-system/src/components/Input/Input.interface.ts:36]: let's remove style prop

File: packages/design-system/src/components/Input/Input.interface.ts [hunk 2]
```diff
@@ -1,85 +1,68 @@
...
+  testId?: string;
+  onIconClick?(): void;
```
Comment (piotrekzawada) [inline packages/design-system/src/components/Input/Input.interface.ts:38]: let's use click event here

File: packages/design-system/src/components/Input/Input.interface.ts [hunk 3]
```diff
@@ -1,85 +1,68 @@
...
+  prefixProps: InputAddonProps;
+  suffixProps: InputAddonProps;
 }
 
-export enum EInputVariants {
+export interface IconProps
+  extends Pick<HTMLAttributes<HTMLElement>, "aria-label"> {
+  icon: TComponentIcon;
+  isError?: boolean;
+  onClick?(): void;
```
Comment (piotrekzawada) [inline packages/design-system/src/components/Input/Input.interface.ts:54]: let's use click event here

File: packages/design-system/src/components/Input/InputComponents.tsx [hunk 4]
```diff
@@ -92,182 +69,170 @@ const InputComponents = {
...
-  }: SuffixIconProps) => {
-    if (!suffixIcon) return <></>;
 
-    const positionStyle = withPrefix ? "col-end-3" : "col-end-2";
+  Icon: ({ icon, isError, onClick, ...props }: IconProps) => {
+    const displayIcon = isError ? (
+      <AlertCircleSvg className="text-input-base-error h-4 w-4" aria-hidden />
+    ) : (
+      <Icon customIcon={icon} className="h-4 w-4" />
+    );
```
[the remark this answers is not in the document]
Comment (piotrekzawada) [reply packages/design-system/src/components/Input/InputComponents.tsx:224]: though i'm not sure if it's good idea 🫠

File: packages/design-system/src/components/InputNumber/InputNumberCounterButton.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import classNames from "classnames";
+import { FC } from "react";
+import { CounterButtonProps } from "./InputNumber.interface";
+
+export const CounterButton: FC<CounterButtonProps> = ({
+  onClick,
+  iconClassName,
+  className,
+  disabled,
+  ...props
```
Comment (monikaracka) [inline packages/design-system/src/components/InputNumber/InputNumberCounterButton.tsx:11]: You don't pass ...props anywhere in this component.

Comment (monikaracka) [inline packages/design-system/src/components/InputNumber/InputNumberCounterButton.tsx:11]: You started to refactor inputs before I added type to buttons. Please check [this PR](https://github.com/elpassion/flounder-2/pull/508/files) and add `types` to buttons.

File: packages/design-system/src/components/InputQuantity/InputQuantityButton.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <button
+      className={classNames(
+        "row-start-2 flex w-min items-center justify-center border-y border-inputQuantity-button bg-inputQuantity-button p-inputQuantity-paddingButton text-inputQuantity-button",
+        "disabled:bg-inputQuantity-button-disabled disabled:text-inputQuantity-button-disabled",
+        "hover:bg-inputQuantity-button-hover active:bg-inputQuantity-button-active",
+        variantStyles[variant]
+      )}
+      style={{ clipPath: "inset(-5px 0px -5px -5px)" }}
```
Comment (monikaracka) [inline packages/design-system/src/components/InputQuantity/InputQuantityButton.tsx:25]: I think you can change this to className.

File: packages/design-system/src/components/Search/index.tsx [hunk 7]
```diff
@@ -7,133 +7,125 @@ import IconButton from "@components/IconButton";
...
     const searchRef = useRef<HTMLInputElement>(null);
-    const isIconVisible = isTyping || !!value;
-    const isDisabled = !!disabled;
-
-    const buttonStyles = classNames(
-      "border-base-button bg-base-button",
-      "hover:border-base-buttonHover hover:bg-base-buttonHover",
-      "active:border-base-buttonActive active:bg-base-buttonActive"
+    console.log(value);
+    console.log(isTyping);
```
Comment (monikaracka) [inline packages/design-system/src/components/Search/index.tsx:33]: to remove

File: packages/design-system/src/components/Search/index.tsx [hunk 8]
```diff
@@ -7,133 +7,125 @@ import IconButton from "@components/IconButton";
 import Input from "@components/Input";
 import type { SearchProps } from "./Search.interface";
 import { GenericKeys } from "@helpers/types";
+import {
+  AddonsContentVariants,
+  InputAddonsVariants,
+} from "@components/Input/Input.interface";
 
 export const Search = forwardRef<HTMLInputElement, SearchProps>(
```
Comment (monikaracka) [inline packages/design-system/src/components/Search/index.tsx:15]: Suffix dropdown shouldn't change color if input has value.
Figma:
Screenshot 2023-08-25 at 13 10 29
Storybook:
Screenshot 2023-08-25 at 13 12 20

**Review by monikaracka — COMMENTED** [review]
I don't know if it's everything. Probably not. You shouldn't have changed the logic in search, refactor InputNumber, InputQuantity, etc. In this task, you should have focused only on the base input and, in other inputs, made only the necessary changes. Then, this PR would be smaller.

**Review by monikaracka — APPROVED** [review]
Great job! 🚀

**Pull request merged**